tests/unit/cluster/slot-migration.tcl

```diff
@@ -355,3 +355,23 @@ start_cluster 3 3 {tags {external:skip cluster} overrides {crash-memcheck-enable
         assert_equal {NOREPLICAS Not enough good replicas to write.} $e
     }
 }
+
+start_cluster 3 3 {tags {external:skip cluster regression} overrides {cluster-allow-replica-migration no cluster-node-timeout 1000} } {
+    # Issue #563 regression test
+    test "Client blocked on XREADGROUP while stream's slot is migrated" {
+        set stream_name aga
+        set slot 609
+
+        # Start a background process to simulate a blocked client on XREADGROUP
+        R 0 XGROUP CREATE $stream_name mygroup $ MKSTREAM
+        exec sh -c "echo 'xreadgroup GROUP mygroup consumer BLOCK 0 streams $stream_name >' | src/valkey-cli -h 127.0.0.1 -p [lindex [R 0 CONFIG GET port] 1] " &
+        after 1000
```

Comment:
Why don't we use a deferring client for this? It seems a lot more reliable than starting up the CLI.